book/src/runtime.md

```diff
@@ -70,14 +81,19 @@ encodes.
 instructions to create memory before passing it to another program.  This
 instruction can be composed into a single transaction with the call to the
 program itself.
+
 2. Runtime guarantees that when memory is assigned to the program it is zero
```

Comment:
You are specifically referring to "CreateAccount"?  Might want to make that explicit that there is no other time the memory is zero'd?

Reply:
what do you mean?  Assign or CreateAccount will enforce that the userdata is zero.

Reply:
Seems strange to zero on assign.  Could see use cases where an account is passed between programs and carrying state with it.

Reply:
@jackcmay you can pass state with a copy.  The program code must accept the copy and actually do it.  Otherwise you can trick the program into an unintended state transition since the program would have no way of distinguishing an "assigned' dirty state, from a state it natively generated.

Reply:
added notes on state security